lib/api/controllers/routerController.js

```diff
@@ -418,15 +215,44 @@ module.exports = function RouterController (kuzzle) {
     api.get('/', (request, response) => {
       var _routes = {};
 
-      extractRouteInfo(_routes, routes);
-      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/_plugin');
+      extractRouteInfo(_routes, routes, '/' + kuzzle.config.apiVersion);
+      extractRouteInfo(_routes, [{verb: 'get', url: '/_serverInfo', controller: 'read', action: 'serverInfo'}]);
+      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/' + kuzzle.config.apiVersion + '/_plugin');
 
       response.setHeader('Access-Control-Allow-Origin', '*');
       response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With');
       response.writeHead(200, {'Content-Type': 'application/json'});
       response.end(stringify({status: 200, error: null, result: {message: 'Available routes for this API version by verb.', routes: _routes}}));
     });
 
+    apiBase.get('/swagger.json', (request, response) => {
+      var content;
+      try {
+        content = fs.readFileSync(__dirname + '/../../../kuzzle-swagger.json');
+      }
+      catch (e) {
+        content = stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500});
```

Comment:
I'm not fond of this solution, because this means that an administrator **must** execute this CLI command for the Swagger route to work.

Since the CLI is using remote actions anyway, wouldn't be a better approach to write the swagger generation code in a core module, returning a promise resolving to a JSON object?
This way, the CLi would still allow generating JSON and YAML files, using this piece of code to generate the swagger object, and this HTTP route would simply call it and return the result.

What do you think?